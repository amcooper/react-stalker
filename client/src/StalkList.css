.StalkList {
  background-color: #07889B;
  color: white;
  list-style-type: none;
}
