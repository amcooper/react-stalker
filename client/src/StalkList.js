import React, { Component } from "react";
import "./StalkList.css";
import StalkListItem from "./StalkListItem";

class StalkList extends Component {
  render() {

    const itemList = this.props.sightings.map(( sighting ) =>
      <li key={sighting.id}><StalkListItem item={sighting} onClick={() => this.props.onClick( sighting.id )} /></li>
    );

    return (
      <ul id="StalkList">
        { itemList }
      </ul>
    )
  }
}

export default StalkList;
