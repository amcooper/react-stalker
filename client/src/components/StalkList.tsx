/// <reference path="./interfaces.d.ts" />

import * as React from "react";
import { Component } from "react";
import "./StalkList.css";
import StalkListItem from "./StalkListItem";

class StalkList extends Component<IStalkListProps, {}> {
  render() {
    const itemList = this.props.sightings.map(sighting => (
      <li key={sighting.id}>
        <StalkListItem
          item={sighting}
          onClick={() => this.props.onClick(sighting.id)}
        />
      </li>
    ));

    return <ul className="StalkList">{itemList}</ul>;
  }
}

export default StalkList;
