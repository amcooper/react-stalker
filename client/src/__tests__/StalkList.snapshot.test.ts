import React from "react";
import StalkList from "../components/StalkList";
import { MemoryRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import { stalkList } from "../fixtures/fixtures";

describe("StalkList component renders the stalk list correctly", () => {
  it("renders correctly", () => {
    const rendered = create(
      <MemoryRouter>
        <StalkList sightings={stalkList} />
      </MemoryRouter>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
