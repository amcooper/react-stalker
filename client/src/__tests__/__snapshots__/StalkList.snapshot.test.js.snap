// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StalkList component renders the stalk list correctly renders correctly 1`] = `
<ul
  className="StalkList"
>
  <li>
    <a
      href="/1"
      onClick={[Function]}
    >
      <p
        className="StalkListItem"
        onClick={[Function]}
      >
        Rosie Perez
         on 
        Mon Apr 23 2018
      </p>
    </a>
  </li>
  <li>
    <a
      href="/2"
      onClick={[Function]}
    >
      <p
        className="StalkListItem"
        onClick={[Function]}
      >
        Angela Bassett
         on 
        Sat Mar 24 2018
      </p>
    </a>
  </li>
  <li>
    <a
      href="/3"
      onClick={[Function]}
    >
      <p
        className="StalkListItem"
        onClick={[Function]}
      >
        Greg Luzinski
         on 
        Sun Feb 25 2018
      </p>
    </a>
  </li>
</ul>
`;
